src/Symfony/Component/DependencyInjection/Compiler/AttributeAutoconfigurationPass.php

```diff
@@ -48,6 +48,13 @@ protected function processValue($value, bool $isRoot = false)
                 $configurator($conditionals, $attribute->newInstance(), $reflector);
             }
         }
+        foreach ($reflector->getMethods() as $method) {
+            foreach ($method->getAttributes() as $attribute) {
+                if ($configurator = $autoconfiguredAttributes[$attribute->getName()] ?? null) {
+                    $configurator($conditionals, $attribute->newInstance(), $method);
```

Comment:
Using all existing callable with methods might have unintended effects if they were expecting to run on classes only. Maybe we should have a separate registration for method-level autoconfiguration attributes

Reply:
That's not really "any callable", but "any callable bound to the attribute", which itself has to opt-in for being accepted on methods. That might be good enough and we might not need more.